cmd/kube-scheduler/app/server.go

```diff
@@ -207,7 +207,11 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 	if cc.DynInformerFactory != nil {
 		cc.DynInformerFactory.WaitForCacheSync(ctx.Done())
 	}
-
+	// Wait for all handlers to sync (all items in the initial list delivered) before scheduling.
+	if err := sched.WaitForHandlersSync(ctx); err != nil {
+		logger.Error(err, "Failed waiting for handlers to sync")
```

Comment:
right, we should return the error here:
```suggestion
		return fmt.Errorf("waiting for handlers to sync: %w", err)
```